sdk/Pulumi.Automation/WorkspaceStack.cs

```diff
@@ -636,6 +638,105 @@ void OnPreviewEvent(EngineEvent @event)
+        /// <returns>The output of the import command, including the generated code, if any.</returns>
+        public async Task<ImportResult> ImportAsync(
+            ImportOptions options,
+            CancellationToken cancellationToken = default)
+        {
+            var args = new List<string>
+            {
+                "import",
+                "--yes",
+                "--skip-preview",
+            };
+
+            var tempDirectoryPath = "";
+            try
+            {
+                // for import operations, generate a temporary directory to store the following:
+                //   - the import file when the user specifies resources to import
+                //   - the output file for the generated code
+                // we use the import file as input to the import command
+                // we the output file to read the generated code and return it to the user
+                tempDirectoryPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+
+                if (options.Resources?.Any() == true)
+                {
+                    Directory.CreateDirectory(tempDirectoryPath);
+                    var importPath = Path.Combine(tempDirectoryPath, "import.json");
+                    var importContent = new
+                    {
+                        nameTable = options.NameTable,
+                        resources = options.Resources,
+                    };
+
+                    var importJson = JsonSerializer.Serialize(importContent, new JsonSerializerOptions
+                    {
+                        DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+                        PropertyNamingPolicy = JsonNamingPolicy.CamelCase
+                    });
+
+                    await File.WriteAllTextAsync(importPath, importJson, cancellationToken);
+                    args.Add("--file");
+                    args.Add(importPath);
+                }
+
+                if (options.Protect is false)
+                {
+                    args.Add("--protect=false");
+                }
+
+                var generatedCodeOutputPath = Path.Combine(tempDirectoryPath, "generated.cs");
+                if (options.GenerateCode is false)
+                {
+                    args.Add("--generate-code=false");
+                }
+                else
+                {
+                    args.Add("--out");
+                    args.Add(generatedCodeOutputPath);
+                }
+
+                if (options.Converter != null)
+                {
+                    // if the user specifies a converter, pass it to `--from <converter>` argument of import.
+                    args.Add("--from");
+                    args.Add(options.Converter);
+                    if (options.ConverterArgs?.Any() == true)
+                    {
+                        // pass any additional arguments to the converter
+                        args.AddRange(options.ConverterArgs);
```

Comment:
Should probably add a "--" arg here to split the converter args from the args to import

Reply:
Done ✅